src/Rings/mpoly-ideals.jl

```diff
@@ -351,12 +351,13 @@ Ideal generated by
 
 ```
 """
-function saturation(I::MPolyIdeal{T}, J::MPolyIdeal{T} = ideal(base_ring(I), gens(base_ring(I))); iteration::Bool=false) where T
-  # `remove` is faster than `Singular.saturation` when saturating with
+function saturation(I::MPolyIdeal{T}, J::MPolyIdeal{T} = ideal(base_ring(I), gens(base_ring(I))); iteration::Bool=false) where {CT<:FieldElem, T <: MPolyRingElem{CT}}
+  # `ppio` is faster than `Singular.saturation` when saturating with
   # respect to principal ideals
-  if ngens(I) == 1 && ngens(J) == 1
-    is_unit(J[1]) && return I
-    return ideal(remove(I[1], J[1])[2])
+  if is_known(is_principal, I) && is_principal(I) && is_known(is_principal, J) && is_principal(J)
```

Comment:
Can't we just test if the corresponding base_ring is a field in order to apply your optimization and let Singular handle the saturation otherwise? Then we would not need to restrict `T` at all.

```suggestion
  if base_ring(I) isa Field && is_known(is_principal, I) && is_principal(I) && is_known(is_principal, J) && is_principal(J)
```